### ISSUE TITLE
Views unformatted lists are not marked up as semantic lists

### ISSUE THREAD

**Issue by msayoung** [localgovdrupal/localgov_base#756]
@Mayur2609 raised this in relation to the Events Listing, but I suspect it affects all of our views listings set to use the unformatted list formatter. 

https://demo.localgovdrupal.org/events

It is potentially a failure of WCAG 1.3.1 : Information, structure, and relationships conveyed through presentation can be programmatically determined or are available in text, e.g. headings, landmarks, lists, tables, footnotes etc.

In this case, the list of content is not marked up as a semantic list. 

As a counter arguement is that having the semantic heading structure does convey the same information as is presented.

Genuinely uncertain about this one - @Mayur2609 can you offer any more background.

I wonder if its been raised in Drupal Core. 

Calling for other thoughts too: @waako @justinepocock @FrazerGNom (or anyone else :) )

**Comment by waako** [comment]
In this specific context, I suspect 1.3.1 is fulfilled by the heading structure, especially as they are the links.

It's nuanced and probably not so bad in this context, but I have noticed Adrian Rosseli (and others) not a fan of headings within list items
> "When lists are lists, hearing "item 4 of 6 heading level 2" is just weird and verbose and why. It is (based on grapevine) part of why VO folks decided not to announce lists when the bullets were styled away -- at that point they weren't generally used as simple lists but instead as holders of richer content." - _Adrian Rosseli_

> Thread summary (relevant parts):
> 1. A list should not contain any bare headings because headings need content. If I understand correctly, this is the case that Adrian considers an abomination and Scott calls "a bit weird".
> 2. A list should not contain some items that are headers+content and other items that are just content, because list items should be parallel […]
>
> source: https://web-a11y.slack.com/archives/C042TSFGN/p1686586449728749?thread_ts=1686231861.080199&cid=C042TSFGN

However, where a views listing teaser contains some of heading, content, and generic action link, I have previously recommended that at bare minimum the unordered list formatter be used instead of unformatted list.
This in order to meet 2.4.4 Link Purpose (In Context) - where no better technique is viable (such as heading no always present or provides insufficient context on it's own).

Although, contrary to my web-a11y quote above, https://demo.localgovdrupal.org/events items are consistent in structure and using unordered list would provide a a helpful count of results to screen-readers, except for VoiceOver of course if styled with `list-style: none` so would need to programmatically add-in the `list` and listitem` roles…

TLDR; In this context 1.3.1 is fulfilled by the heading structure, but possibly using semantic lists would do no harm

**Comment by markconroy** [comment]
I think Drupal by default should use HTML list as the default formatter for views, but it doesn't.

I wonder should we consider 
- rewriting the unformatted list template to use`ul/ol` instead of `div` or
- rebuilding our views to use HTML list or 
- something else

I know on other projects, I rewrote the unformatted list template to use `ol` and then if a view (which is a listing) didn't want to be a list for some reason - we could add a custom template for it.
